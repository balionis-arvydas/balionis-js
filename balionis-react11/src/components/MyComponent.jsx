import React, { Component } from "react";
import "./MyComponent.less";

class MyComponent extends Component {
    constructor () {
        super();

        this.state = {
            value: "Hello World."
        };
    }

    render () {
        return (
            <p className="myComponentWrapper">{this.state.value}</p>
        );
    }
}

export default MyComponent;
