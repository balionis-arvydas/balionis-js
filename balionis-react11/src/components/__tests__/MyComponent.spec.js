import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MyComponent from '../MyComponent';

describe('MyComponent', () => {
    test("snapshot", async () => {
        const component = renderer.create(
            <MyComponent/>,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("message", async () => {
        const component = shallow(<MyComponent/>);
        expect(component.text()).toEqual('Hello World.');
    });
});
