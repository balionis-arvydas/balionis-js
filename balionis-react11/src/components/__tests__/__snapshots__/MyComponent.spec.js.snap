// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent snapshot 1`] = `
<p
  className="myComponentWrapper"
>
  Hello World.
</p>
`;
